opm/grid/cpgrid/PartitionTypeIndicator.cpp

```diff
@@ -9,21 +9,13 @@ namespace Dune
 {
 namespace cpgrid
 {
-PartitionType PartitionTypeIndicator::getPartitionType(const EntityRep<0>& cell_entity) const
+PartitionType PartitionTypeIndicator::getPartitionType(const Entity<0>& cell_entity) const
 {
     if(cell_indicator_.size())
         return PartitionType(cell_indicator_[cell_entity.index()]);
-    return InteriorEntity;
-}
-
-PartitionType PartitionTypeIndicator::getPartitionTypeWhenLgrs(const Entity<0>& cell_entity, bool lgrsOnDistributedGrid) const
-{
-    if(cell_indicator_.size())
-        return PartitionType(cell_indicator_[cell_entity.index()]);
-    // Assuming grid has been distributed and some LGRs have been added afterwards.
-    // For level 0, it's defined. For other levels, the refined cell inherits its father
-    // attribute.
-    if((grid_data_->level_data_ptr_->size()>1) && lgrsOnDistributedGrid) {
+    // When level zero grid has been distributed and some LGRs have been added afterwards, refined cells
+    // inherit its parent cell attribute.
+    if (grid_data_->level_ >0) { // level_ > 0 only for refined level grids.
```

Comment:
Good, this will work as long as we have the whole hierarchy on one process. That is the case now.

We just have to keep this in mind.

Reply:
Thanks! That was fast! :)